staging/src/k8s.io/api/resource/v1alpha2/types.go

```diff
@@ -347,6 +407,11 @@ type ResourceClass struct {
 	// Setting this field is optional. If null, all nodes are candidates.
 	// +optional
 	SuitableNodes *v1.NodeSelector `json:"suitableNodes,omitempty" protobuf:"bytes,4,opt,name=suitableNodes"`
+
+	// If and only if allocation of claims using this class is handled
+	// via structured parameters, then StructuredParameters must be set to true.
```

Comment:
wouldn't this be deductible from the claim?

Reply:
No, the information in the claim is not sufficient. The parameter reference could be for a vendor CRD, and then the scheduler wouldn't know whether it needs to wait for a corresponding `ResourceClaimParameters` object (structured parameters) or use `PodSchedulingContext` to communicate with a driver controller ("classic" DRA).

Reply:
Can we make it a pointer to a boolean?

I picture a world where the default for this field is true and you have to explicitly opt out.

cc @johnbelamaric

Reply:
I think that could makes sense, the goal would be for this to be the most common case.

Reply:
Okay. I'll make it a pointer, but for now will keep the default at false because we are not breaking existing DRA drivers in 1.30.

Reply:
That's acceptable, but we should probably do so for 1.31